Sources/NetworkRequestHelper.swift

```diff
@@ -21,42 +21,73 @@ import XCTest
 ///    - ``MockNetworkService``
 ///    - ``RealNetworkService``
 class NetworkRequestHelper {
-    private(set) var orderedNetworkRequests: [NetworkRequest] = []
-    private var sentNetworkRequests: [TestableNetworkRequest: [NetworkRequest]] = [:]
+    private let queue: DispatchQueue = .init(label: "com.adobe.testutils.networkrequesthelper.queue")
+
+    private var _orderedNetworkRequests: [NetworkRequest] = []
     /// Matches sent `NetworkRequest`s with their corresponding `HttpConnection` responses.
-    private(set) var networkResponses: [TestableNetworkRequest: [HttpConnection]] = [:]
+    private var _networkResponses: [TestableNetworkRequest: [HttpConnection]] = [:]
+
+    private var sentNetworkRequests: [TestableNetworkRequest: [NetworkRequest]] = [:]
     private var expectedNetworkRequests: [TestableNetworkRequest: CountDownLatch] = [:]
 
+    var orderedNetworkRequests: [NetworkRequest] {
+        return queue.sync {
+            return _orderedNetworkRequests.map { $0.deepCopy() }
+        }
+    }
+
+    var networkResponses: [TestableNetworkRequest: [HttpConnection]] {
+        return queue.sync {
+            var copiedResponses: [TestableNetworkRequest: [HttpConnection]] = [:]
+            for (key, value) in _networkResponses {
+                copiedResponses[key] = value.map { $0.deepCopy() }
+            }
+            return copiedResponses
```

Comment:
Since you are not modifying the keys, you can reduce this to 
```
return _networkResponses.mapValues { value in 
    return value.map { $0.deepCopy() }
}
```